client/components/FaveButton.tsx

```diff
@@ -1,24 +1,42 @@
 import React from "react";
-import { Fave } from "@prisma/client";
 import { ZoopWithDetails } from "../../src/types/custom";
 
 import StarBorderIcon from "@mui/icons-material/StarBorder";
 import Typography from "@mui/material/Typography";
 import Stack from "@mui/material/Stack";
 import Button from "@mui/material/Button";
 
+import { useAddFaveMutation} from "../features/api";
+import { useSelector } from "react-redux";
+import { RootState } from "../app/store";
+
 type FaveButtonProps = {
   zoop: ZoopWithDetails
 };
 
 const FaveButton = ({ zoop }: FaveButtonProps) => {
 
+  const currentUser = useSelector((state: RootState) => state.auth.user)
+
+  const [addFave, {isLoading, data, error}] = useAddFaveMutation();
+
+
   const handleClick = ((e: React.MouseEvent<HTMLButtonElement, MouseEvent>) => {
     e.preventDefault();
+    if (!isLoading && currentUser) {
+      addFave({
+        faverId: currentUser.id,
+        zoopId: zoop.id
+      })
+    }
   })
   
   return (
-      <Button variant="outlined" onClick={(e) => handleClick(e)}>
+      <Button 
+        variant="outlined"
+        disabled={!currentUser || currentUser.id === zoop.authorId}  
```

Comment:
Might be wise to show some tooltip if disabled to explain why

Reply:
I'll add an issue